### PR TITLE
Use flake8 to find and fix undefined names

### PR DESCRIPTION
**Pull request by cclauss**
$ __python3 -m flake8 . --count --select=E9,F82 --show-source --statistics__

This is a great way to find syntax errors and undefined names in a code base when ensuring compatibility with both Python 2 and Python 3.  This PR uses this approach to find and fix issues that have the potential to raise __NameError__ at runtime.

### CODE REVIEW

File: scapy/modules/krack/crypto.py [hunk 1]
```diff
@@ -4,10 +4,12 @@
 from struct import unpack, pack
 from zlib import crc32
 
-from cryptography.hazmat.primitives.ciphers import Cipher, algorithms, modes
+from cryptography.hazmat.primitives.ciphers import Cipher, algorithms
 from cryptography.hazmat.backends import default_backend
 
 from scapy.compat import hex_bytes, orb
+from scapy.modules.six import integer_types
+from scapy.modules.six.moves import xrange
```
Comment (gpotter2) [inline scapy/modules/krack/crypto.py]: Shouldn’t it be range ? you are replacing xrange by range below

File: scapy/modules/krack/crypto.py [hunk 2]
```diff
@@ -144,7 +146,7 @@ def gen_TKIP_RC4_key(TSC, TA, TK):
     assert len(TSC) == 6
     assert len(TA) == 6
     assert len(TK) == 16
-    assert all(isinstance(x, (int, long)) for x in TSC + TA + TK)
+    assert all(isinstance(x, integer_types) for x in TSC + TA + TK)
```
Comment (gpotter2) [inline scapy/modules/krack/crypto.py]: For this, we have used every where else in scapy a static call:
ˋimport scapy.modules.six as six`

Then directly
`six.integer_types`

It’s not needed but would be nice to keep code consistency

File: scapy/arch/common.py [hunk 3]
```diff
@@ -15,6 +15,11 @@
 from scapy.config import conf
 import scapy.modules.six as six
 
+try:
+    long        # Python 2
+except NameError:
+    long = int  # Python 3
```
Comment (gpotter2) [inline scapy/arch/common.py]: long is used in a compatibility function only used with Python 2. This should not be needed

File: scapy/layers/tls/tools.py [hunk 4]
```diff
@@ -176,7 +177,7 @@ def _tls_aead_auth_encrypt(alg, p, write_seq_num):
     write_seq_num = struct.pack("!Q", write_seq_num)
```
Comment (gpotter2) [inline scapy/layers/tls/tools.py]: It’s not related to your PR, but I just noticed it. Would you mind replacing the str (line above this comment)
by a `raw()` call (import if needed from scapy.compat).
Would be real nice

Comment (cclauss) [reply scapy/layers/tls/tools.py]: Done.

**Comment by cclauss** [comment]
On the spaces thing...  Sorry about that but when saving my editor automatically strips trailing whitespace and ensures there is one and only one \n at the end of the file.

In Python 2, __xrange()__ is an iterator while __range()__ _builds a list in memory_ and the steps through it.  There are both memory and performance penalties for choosing __range()__ when the range is big.  I assume that the Python 2 programmer wanted an iterator when they chose __xrange()__ so we should not undo that unless we know that the number is so small (like 6) that there is no noticeable difference.  When we know for certain that there is no difference then we should move away from the legacy syntax.
```
import sys
print(sys.getsizeof(xrange(6)))     #    40
print(sys.getsizeof(range(6)))      #   120
print(sys.getsizeof(xrange(6000)))  #    40
print(sys.getsizeof(range(6000)))   # 48072
```

**Comment by gpotter2** [comment]
I know the difference :)
I only meant that according to https://pythonhosted.org/six/
six.moves.xrange and six.moves.range point to the same thing, so the import coulas have been renamed.

You implémentation is good :)

File: scapy/layers/tls/tools.py [hunk 5]
```diff
@@ -208,4 +210,3 @@ def _tls_aead_auth_decrypt(alg, c, read_seq_num):
     if p.fragment is None: # Verification failed.
         return None
     return p
-
```
Comment (guedou) [inline scapy/layers/tls/tools.py]: Please remove useless changes.

File: scapy/modules/six.py [hunk 6]
```diff
@@ -23,6 +23,8 @@
 ## Copyright (C) Philippe Biondi <phil@secdev.org>
 ## This program is published under a GPLv2 license
 
+# flake8: noqa: F821
```
Comment (guedou) [inline scapy/modules/six.py]: Please remove this.

File: scapy/plist.py [hunk 7]
```diff
@@ -10,13 +10,13 @@
 
 from __future__ import absolute_import
 from __future__ import print_function
-import os,subprocess
+import os, subprocess
```
Comment (guedou) [inline scapy/plist.py]: All changes in this file are useless (regarding bugs), please remove them.

File: scapy/modules/six.py (hunk 6 above)
Comment (cclauss) [reply scapy/modules/six.py]: Done.

**Comment by cclauss** [comment]
It would be easier to contribute to this project if you could just do a one time:
* __autopep8 --in-place --recursive .__  # or yapf is you like
* __isort -rc --atomic .__

You can then have Flake8-on-Travis act the automatic den mother on all style violations in future PRs.  They they have style violations, they fail the build and do not get reviewed (like Django does).

More and more I understand and like the Golang approach of a single code format with a formatter that is built into the language tooling.

**Comment by guedou** [comment]
This also adds drawbacks, including breaking code history. linting will definitely happen (hopefully in 2018), but please let us do that at your own pace =)

We do enforce PEP-08 for new code.

**Comment by cclauss** [comment]
Do I need to be worried about the AppVeyor results?

**Comment by gpotter2** [comment]
I wonder if could be possible to create a script that calls autopep8, then iterate trough each changed line and edit the last commit changing those lines

**Comment by cclauss** [comment]
Ready for review.

**Comment by cclauss** [comment]
;-)  https://github.com/ambv/black

**Comment by cclauss** [comment]
@guedou Could you please let me know on the failing tests?  I do not know how to interpret them.

**Comment by guedou** [comment]
@cclauss I restarted the test and the results look OK now.

File: scapy/main.py [hunk 8]
```diff
@@ -199,8 +201,9 @@ def list_contrib(name=None):
 def update_ipython_session(session):
     """Updates IPython session with a custom one"""
     try:
+        global get_ipython  # defined externally by ipython / jupyter
         get_ipython().user_ns.update(session)
-    except:
+    except Exception:
```
Comment (guedou) [inline scapy/main.py]: This change is unnecessary for this PR. Could you remove it?

Comment (cclauss) [reply scapy/main.py]: “We do enforce PEP8 for new code.”

**Comment by cclauss** [comment]
@gpotter2  I should have listened to [your warning](https://github.com/secdev/scapy/pull/1250#pullrequestreview-104011209). Scapy is closed source.

**Comment by guedou** [comment]
This kind of behaviour is not helpful. I spent quite some time since this weekend to write a proposal to apply PEP-08 to Scapy in a way that will be useful and fair to everyone. Hopefully, I will be able to post it this week.

I am only asking that you remove few bytes from your patch that I am willing to merge as I find it useful for the project. If you really want to catch exceptions, you can propose a patch that will correctly catch exceptions raised by the block.

**Comment by cclauss** [comment]
_You are modifying a line already present in Scapy without any effect on its behavior._ 

Please reread PEP8 to understand why the proposed change _does have an effect_ and is recommended.

**Comment by gpotter2** [comment]
@cclauss Hi,
I fully support guedou on this subject. We all know what PEP08 is, and would like to see it more in scapy's codebase.
However, if we want to keep the git history clean, and avoid breaking `git blame` (which is used a lot), it's not possible to have already in-place code replaced for this only reason.

If you edit some lines, it's very recommanded if not compulsory to update them to PEP08, but otherwise it's not a good point. A such big repository would become impossible to maintain.

It may be frustrating to have comments on your PR, and we are aware of it. Moreover, scapy 2.4.0 is out tommorow so all mainteners currently are finishing what needs to be done, and have no time to waste.

If you could cancel the PEP08 changes, it would be really nice of you

**Pull request closed without merging**